app/src/server/utils/convertFunctionCalls.ts

```diff
@@ -0,0 +1,96 @@
+import type { ChatCompletionCreateParamsBase } from "openai/resources/chat/completions";
+
+export const convertFunctionCall = (
+  functionCall: ChatCompletionCreateParamsBase["function_call"] | null,
+): ChatCompletionCreateParamsBase["tool_choice"] => {
+  if (!functionCall) {
+    return undefined;
```

Comment:
This seems a little concerning to me, in the sense that it could take a null and return an undefined, which could lead to unexpected errors. I'd expect it to return the same thing it got if it isn't a valid function call, or at least to be explicit about the conversion in the type signature and/or comments.

Reply:
My intent was to allow all callers to rely on `convertFunctions` returning a value that matches the `tool_choice` type, but I see how that could be confusing. I'll modify the type signature and return null as well.